devito/dse/rewriters.py

```diff
@@ -406,6 +408,38 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        A pass to perform only the basic skewing in order to have valid data
+        dependences.
+        """
+        skew_factor = configuration['skew_factor']
+        skew_dim, mapper, int_mapper = None, {}, []
+        total_int = {}
+        intervals, sub_iterators, directions = cluster.ispace.args
+
+        for dim in cluster.ispace.dimensions:
+            if isinstance(dim, SpaceDimension):
+                mapper[dim] = dim + skew_factor*skew_dim
+                int_mapper.append(IntervalGroup([Interval(dim, -skew_factor*skew_dim,
+                                                -skew_factor*skew_dim)]))
+            elif dim.is_Time:
+                if isinstance(dim, TimeDimension):
+                    int_mapper.append(IntervalGroup([Interval(dim, 0, 0)]))
+                    skew_dim = dim
+
+        if skew_dim is None:
+            return cluster
+
+        total_int = IntervalGroup.generate('union', *int_mapper)
```

Comment:
this variable, for homogeneity with the rest of the codebase, should be called just ``intervals``